src/backend/distributed/commands/alter_table.c

```diff
@@ -1372,6 +1372,21 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 															schemaName, targetName);
 			SendCommandToWorkersWithMetadata(workerChangeSequenceDependencyCommand);
 		}
+		else if (ShouldSyncTableMetadata(sourceId))
+		{
+			/*
+			 * We are converting a citus local table to a distributed/reference table,
```

Comment:
As I understand, we hit that check if we are changing the shard count of the table as well. Can we update the comment accordingly to be more explicit about the reasoning?

Reply:
I think we hit `if (ShouldSyncTableMetadata(targetId))` for those cases, not this else part.